### PR TITLE
add socket and threads in templates and dedicated cpu

### PR DESCRIPTION
**Pull request by ksimon1**
Added notice about sockets and threads in common templates and dedicated cpu section. 
@MarSik, @fabiand please can you review? Please feel free to point out, where the notice about sockets and threads is missing.

### CODE REVIEW

File: creating-virtual-machines/dedicated-cpu.adoc [hunk 1]
```diff
@@ -42,8 +42,8 @@ pinned to the pCPUS that has been dedicated for the virt-launcher
 container.
 
 Expressing the desired amount of VMI’s vCPUs can be done by either
-setting the `spec.domain.cpu.cores` or
-`spec.domain.resources.requests.cpu` or
+setting the `spec.domain.cpu.sockets` or `spec.domain.cpu.cores` or
+ `spec.domain.cpu.threads` or `spec.domain.resources.requests.cpu` or
 `spec.domain.resources.limits.cpu` to a whole number, integer (e.g. 1,
 2, etc) indicating the number of vCPUs requested for the VMI.
```
Comment (MarSik) [inline creating-virtual-machines/dedicated-cpu.adoc]: The notes below should probably be updated as well. (to sockets*cores*threads == to resources.cpu)

File: creating-virtual-machines/dedicated-cpu.adoc [hunk 2]
```diff
@@ -42,8 +42,8 @@ pinned to the pCPUS that has been dedicated for the virt-launcher
 container.
 
 Expressing the desired amount of VMI’s vCPUs can be done by either
-setting the `spec.domain.cpu.cores` or
-`spec.domain.resources.requests.cpu` or
+setting the `spec.domain.cpu.sockets` or `spec.domain.cpu.cores` or
```
Comment (MarSik) [inline creating-virtual-machines/dedicated-cpu.adoc]: Please explain the topology a bit more. I would even say:

```suggestion
setting the guest topology in `spec.domain.cpu` (`sockets`, `cores`, `threads`) or resources (requests.cpu, limits.cpu).
```

File: creating-virtual-machines/dedicated-cpu.adoc [hunk 3]
```diff
@@ -64,7 +64,9 @@ kind: VirtualMachineInstance
 spec:
   domain:
     cpu:
+      sockets: 1
```
Comment (MarSik) [inline creating-virtual-machines/dedicated-cpu.adoc]: Please update the example so it prefers sockets.

File: creating-virtual-machines/dedicated-cpu.adoc (hunk 1 above)
Comment (MarSik) [reply creating-virtual-machines/dedicated-cpu.adoc]: Also add a Note that will explain the performance advantage of using sockets.

**Comment by fabiand** [comment]
To me this change looks good, awaiting the kubevirt change to be merged.

**Comment by ksimon1** [comment]
@MarSik, @fabiand 
[PR](https://github.com/kubevirt/kubevirt/pull/1839) in kubevirt repo is merged, now this can be merged as well.

**Comment by fabiand** [comment]
Thanks

**Pull request merged**